### DISCUSSION TITLE
Embedded Athens

### DISCUSSION THREAD

**Discussion by josephjacks** [athensresearch/athens#609] [General]
Is it possible to use Athens as a library? For example, I'd love to see Athens as a core runtime feature of Forem: https://github.com/forem/forem

**Comment by jefftangx** [comment]
That's an interesting idea! Especially if the embedded Athens was a shared database.

But what if Forem just had bi-directional links as a primitive? @josephjacks

**Comment by josephjacks** [comment]
> That's an interesting idea! Especially if the embedded Athens was a shared database.
> 
> 
> 
> But what if Forem just had bi-directional links as a primitive? @josephjacks

@tangjeff0 

perhaps two options:

1) build a separate embedded version of Athens without a built-in database but instead hooks to integrate inside any existing multi-tenant CMS-like application to include all the core primitive functional aspects of Athens without the front end UI, etc., without needing its own database. This would require a rethink of not too tightly coupling Athens itself to DataScript. I feel like this would require the complete decoupling of app logic from the database (DataScript). You'd then need some clean universal way to script together an integration into any kind of database or maybe just 2-3 common reference implementations.

or 2) Different core Athens primitives could be made available as decoupled modules or libs that any other CMS-oriented app would take a dependency on and build around, but those modules would require DataScript as a dependency as well, so the user of this kind of Embedded Athens would need to design their app to suits this architecture.

Or something in between.

**Comment by benhalpern** [comment]
I'm all for exploring this idea! One way or another the bi-directional links are a great idea. All neat ideas @josephjacks.

@tangjeff0 _Technically_ I have no idea how Athens could fit in but I'm very happy and willing to learn as much as I can to best evaluate this.
